src/Model/Item.php

```diff
@@ -132,6 +134,12 @@ public static function deleteById($item_id, $priority = PRIORITY_HIGH)
 			$parent = ['origin' => false];
 		}
 
+		// "Deleting" global items just means hiding them
+		if (($item['uid'] == 0) && ($uid != 0)) {
+			dba::update('user-item', ['hidden' => true], ['iid' => $item_id, 'uid' => $uid], true);
+			return true;
```

Comment:
Nope, no, no no no, not at all. Method is called `Item::deleteById` and it should delete an item by id. You need to make a separate method to hide global items.

Reply:
This comment stands no matter what.

Reply:
If we would use separate functions, we had to add switches for deciding between "hiding" and "deletion" at every existing call of Item::deleteByID.

Reply:
That's what I want. You can make a `Item::deleteOrHide()` function if you want but you can't leave this `user-item` query in `deleteById`.

Reply:
I personally think that it is some kind of deletion anyway, since it removes items from the user's stream. And with the upcoming changes to the item class there will be even more abstraction.

Reply:
It "removes", it "hides", but it doesn't "delete", and I don't want to keep this confusion. Please create a separate according method.

Reply:
I guess that in a future version of the class this function will not simply set this "hidden" variable in the "user-item" table but will delete the "user-item" table entry - or the "item" table entry, if it was a user's origin post.

Reply:
Yes, and when it happens, we’ll talk about it. For now, it’s a blocker for me. No cruft.

Reply:
Okay.